.github/PULL_REQUEST_TEMPLATE

```diff
@@ -1,10 +1,32 @@
-## What changes were proposed in this pull request?
+<!--  Thanks for sending a pull request!  Here are some tips for you:
+1. If this is your first time, please read our contributor guidelines: https://spark.apache.org/contributing.html
+2. Ensure you have added or run the appropriate tests for your PR: https://spark.apache.org/developer-tools.html
+3. Understand the relationship beteween JIRAs to file issues and PRs to propose some changes for them, as documented in guidelines.
+4. If the PR is unfinished, add '[WIP]' in your PR title, e.g., '[WIP][SPARK-XXXX] Your PR title ...'
+5. Update your PR description whenever it is needed.
+6. Provide a minimised example to reproduce the issue if possible.
+-->
 
-(Please fill in changes proposed in this fix)
+### What changes were proposed in this pull request?
 
-## How was this patch tested?
+### Why changes are needed?
 
-(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
-(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)
+### Special notes for your reviewer
```

Comment:
I am note sure about this section. I mean, any note can also be left in the other previous sections (e.g a link to a design doc may be put in the `What changes...` and a reference to a mailing list discussion in the `Why changes...`). I feel like this section will we always left empty and the times it'd be filled, we can put the same info in other sections.

Reply:
I think the problem is that all the sections here can be left in `What changes...` or `Why changes...` but people don't do it. I would like to encourage contributors to pay attentions to leave some notes for easier and faster review..

Reply:
mmmh...then what about giving more suggestions on how to make a review easier to be reviewed instead of creating a dedicated section?

Reply:
+1 on @mgaido91 , and maybe in many cases adding self comments on changeset would be more helpful. I'd rather not expect people don't explain enough on `What changes...` and `Why changes...` but try to explain this section.

Reply:
I think most of the time this section will be empty. If a contributor does have some special notes, he usually will find a way to put it, e.g. at the end of the PR description, or add a new comment.